src/main/java/bob/Storage.java

```diff
@@ -0,0 +1,81 @@
+package bob;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.PrintWriter;
+import java.io.UnsupportedEncodingException;
+import java.time.format.DateTimeParseException;
+import java.util.Objects;
+import java.util.Scanner;
+
+import bob.exceptions.EmptyArgumentException;
+import bob.exceptions.MissingArgumentException;
+import bob.tasks.Task;
+import bob.tasks.TaskList;
+
+/**
+ * Handles all the storage related actions
+ */
+public class Storage {
+
+    /**
+     * Reads the data from the given file and adds it to the taskList
+     *
+     * @param taskList TaskList to which all the data is to be added into
+     * @param filePath File path to the file from which the data is to be read from
+     */
+    public static void readData(TaskList taskList, String filePath) {
+        try (Scanner scanner = new Scanner(new File(filePath), "UTF-8")) {
+            while (scanner.hasNextLine()) {
+                String in = scanner.nextLine();
+                try {
+                    Task newTask;
+                    switch (in.split(" ")[0]) {
```

Comment:
Well-followed code style! I can see the proper use of Egyptian brackets and proper indentations as well.